src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -299,8 +299,8 @@ export abstract class NotebookInput extends EditorInput {
 	}
 
 	private async setTrustForNewEditor(newInput: IEditorInput | undefined): Promise<void> {
-		let isTrusted = this._model.getNotebookModel().trustedMode;
-		if (isTrusted && newInput && newInput.resource !== this.resource) {
+		let model = this._model.getNotebookModel();
```

Comment:
should we await the notebook model being ready and then serialize the change instead of skipping this?

Reply:
We don't wait on model load elsewhere, like in notebookInput.updateModel. We skip model changes if it's undefined.

Reply:
i guess the behavior question is... should we in this case? User presses save right after ADS loads. Can we help them out here?

Reply:
Maybe, but this is a broader issue. For example, if you try to modify a file and save it right away, the changes aren't persisted if you close and reopen. The file doesn't even show up as dirty.

Reply:
that's true. ok let's go with this.